core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/deploysemantics/locationpreference/LocationPreference.scala

```diff
@@ -10,36 +9,8 @@ case class AddressInfo(
     controllerAddress: Address
 )
 
-trait LocationPreference extends Serializable {
+sealed trait LocationPreference extends Serializable
 
-  def getPreferredLocation(
-      addressInfo: AddressInfo,
-      physicalOp: PhysicalOp,
-      workerIndex: Int
-  ): Address
+object PreferController extends LocationPreference
```

Comment:
Although it is clear to me, please add some comment and example to explain this preference.

Reply:
done